examples/nlp/gpt/train_gpt_dpo.py

```diff
@@ -138,7 +138,7 @@ def main(cfg) -> None:
 
     logger.log_hyperparams(OmegaConf.to_container(cfg))
 
-    timer = Timer(cfg.exp_manager.get("max_time_per_run"))
+    timer = Timer(cfg.exp_manager.get("max_time_per_run") if cfg.exp_manager else None)
```

Comment:
why do you need to guard this? shouldn't it always exist(or is this because of the CI)

Reply:
So there's occasional race conditions when launching with mpirun + experiment manager. I had encountered issues where several ranks will try to os.makedir and fail because the later one errors b/c it already exists

`cfg.exp_manager is None` is valid, so I added this change everywhere since we should be allowed to run without the experiment manager, but I just needed https://github.com/NVIDIA/NeMo/pull/10651

Reply:
okay this change is fine with me. but shouldn't nemo just guard it themselves? seems like a serious bug in pretty hot piece of code